src/main/java/duke/Storage.java

```diff
@@ -0,0 +1,80 @@
+package duke;
+
+import java.io.File;
+import java.io.FileWriter;
+import java.io.IOException;
+import java.util.Scanner;
+
+import duke.command.Command;
+import duke.command.Parser;
+import duke.task.Task;
+import duke.task.TaskList;
+import duke.ui.UI;
+
+/**
+ * Storage handler for duke application.
+ */
+public class Storage {
+
+    private final String savePath;
+    private final boolean isUpdatable = true;
+
+    public Storage(String savePath) {
+        this.savePath = savePath;
+    }
+
+    /**
+     * Updates the savefile to the most current tasks.
+     *
+     * @param taskList the current list of tasks
+     */
+    public void updateFile(TaskList taskList, UI ui) {
```

Comment:
This is a very neatly written function, the try catch logic is clear and explicit, making the code very readable 👍